### PR TITLE
DOP-3939: Return metadata for same branch name in different projects

### PR DESCRIPTION
**Pull request by rayangler**
### Ticket

DOP-3939

### Notes

* Updates the metadata aggregation to ensure that different projects with the same branch name are actually returned.

### CODE REVIEW

File: tests/sampleData/metadata.ts [hunk 1]
```diff
@@ -101,4 +101,34 @@ export const sampleMetadata = [
...
+    title: 'MongoDB Documentation',
+    eol: false,
+    slugToTitle: {},
+    toctreeorder: {},
+    parentPaths: {},
+    static_files: {},
+    build_id: new ObjectId('642ec854c38bedd45ed31238'),
+    created_at: new Date('2023-04-07T19:00:00.000Z'),
+    updated_at: new Date('2023-04-08T14:00:00.000Z'),
+  },
```
Comment (mmeigs) [inline tests/sampleData/metadata.ts:118]: Maybe overkill, but could we add a metadata document that has the same `project` and `branch` name as one of these, but a different user?

**Review by mmeigs — APPROVED** [review]
LGTM! So glad you caught this.

File: src/services/database.ts [hunk 2]
```diff
@@ -135,13 +135,13 @@ export const findLatestMetadataByProperty = (filter: Filter<Document>, req: Requ
...
-    // Look for all metadata documents of the same project
+    // Look for all metadata documents that match the filter
     { $match: filter },
     // Sort them so that most recent documents are first
     { $sort: { created_at: -1 } },
-    // Group documents by their branch, and only embed the first doc seen
+    // Group documents by their branch and project, and only embed the first doc seen
     // (or most recent, based on sorting stage)
-    { $group: { _id: '$branch', doc: { $first: '$$ROOT' } } },
+    { $group: { _id: { project: '$project', branch: '$branch', user: '$github_username' }, doc: { $first: '$$ROOT' } } },
```
[the remark this answers is not in the document]
Comment (seungpark) [reply src/services/database.ts]: FWIW i think this was before we had routes per user

**Review by seungpark — APPROVED** [review]
LGTM !

File: tests/sampleData/metadata.ts (hunk 1 above)
Comment (rayangler) [reply tests/sampleData/metadata.ts:118]: Done! Tests seem to pass without any snapshot updates. The new test should have caught the second `landing` project if it was an issue 👍

**Pull request merged**